### PR TITLE
inventory fixes

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

Avoid incomplete updates

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

inventory
##### ANSIBLE VERSION

```
2.5/2.4
```

### CODE REVIEW

**Comment by abadger** [comment]
shipit

File: lib/ansible/inventory/data.py [hunk 1]
```diff
@@ -67,11 +67,24 @@ def __init__(self):
...
 
     def serialize(self):
-        data = dict()
+        self.cache = None
+        self._groups_dict_cache = None
+        data = {
+            groups = self.groups
+            hosts = self.hosts
+            localhost = self.localhost
+            current = self.current_source
```
Comment (mattclay) [inline lib/ansible/inventory/data.py]: CI failure due to syntax error here.

**Comment by mattclay** [comment]
CI failure in integration tests due to a traceback on Ubuntu 14.04, Centos 6, Centos 7 and RHEL 7.4:

```
2017-09-26 15:48:34 Traceback (most recent call last):
2017-09-26 15:48:34   File "/usr/lib/python2.7/multiprocessing/queues.py", line 266, in _feed
2017-09-26 15:48:34     send(obj)
2017-09-26 15:48:34   File "/root/ansible/lib/ansible/plugins/cache/memory.py", line 50, in __getstate__
2017-09-26 15:48:34     return self.copy()
2017-09-26 15:48:34   File "/root/ansible/lib/ansible/plugins/cache/memory.py", line 47, in copy
2017-09-26 15:48:34     return self._cache.copy()
2017-09-26 15:48:34 AttributeError: 'CacheModule' object has no attribute '_cache'
```

**Comment by mattclay** [comment]
CI failure due to traceback in integration tests:

```
2017-10-13 03:27:48 Traceback (most recent call last):
2017-10-13 03:27:48   File "/usr/lib64/python2.7/multiprocessing/queues.py", line 266, in _feed
2017-10-13 03:27:48     send(obj)
2017-10-13 03:27:48   File "/root/ansible/lib/ansible/plugins/cache/memory.py", line 50, in __getstate__
2017-10-13 03:27:48     return self.copy()
2017-10-13 03:27:48   File "/root/ansible/lib/ansible/plugins/cache/memory.py", line 47, in copy
2017-10-13 03:27:48     return self._cache.copy()
2017-10-13 03:27:48 AttributeError: 'CacheModule' object has no attribute '_cache'
```

**Comment by mattclay** [comment]
CI failure due to traceback:

```
2017-10-13 21:39:44 Traceback (most recent call last):
2017-10-13 21:39:44   File "/usr/lib64/python2.6/multiprocessing/queues.py", line 242, in _feed
2017-10-13 21:39:44     send(obj)
2017-10-13 21:39:44   File "/root/ansible/lib/ansible/plugins/cache/memory.py", line 50, in __getstate__
2017-10-13 21:39:44     return self.copy()
2017-10-13 21:39:44   File "/root/ansible/lib/ansible/plugins/cache/memory.py", line 47, in copy
2017-10-13 21:39:44     return self._cache.copy()
2017-10-13 21:39:44 AttributeError: 'CacheModule' object has no attribute '_cache'
```

**Comment by mattclay** [comment]
The failure is causing CI jobs to hang and eventually time out, so I canceled the last run once it reached that point.

**Comment by mattclay** [comment]
Syntax errors are valid. The opensuse failures are unrelated.

**Pull request merged**